lib/internal/http2/core.js

```diff
@@ -2306,30 +2304,11 @@ class Http2Stream extends Duplex {
     trackWriteState(this, req.bytes);
 
     if (this.session[kType] === NGHTTP2_SESSION_CLIENT && onClientStreamBodyChunkSentChannel.hasSubscribers) {
-      let chunk;
-
-      if (ArrayIsArray(data)) {
-        const buffers = [];
-        for (let i = 0; i < data.length; ++i) {
-          if (typeof data[i] === 'object') {
-            if (typeof data[i].chunk === 'string') {
-              buffers.push(Buffer.from(data[i].chunk, data[i].encoding));
-            } else {
-              buffers.push(data[i].chunk);
-            }
-          }
-        }
-
-        chunk = Buffer.concat(buffers);
-      } else if (typeof data === 'string') {
-        chunk = Buffer.from(data);
-      } else {
-        chunk = data;
-      }
-
       onClientStreamBodyChunkSentChannel.publish({
         stream: this,
-        chunk,
```

Comment:
The event contents of a diagnostics_channel should be considered public API surface, therefore removing/replacing this would be a breaking change. Perhaps we could add a getter to the message that does the prior logic lazily, with a one-time deprecation warning?

Reply:
Ah, seems so. The GitHub mobile app sometimes seems to do that with notifications for me. 😅